lib/shoehorn/application/event_supervisor.ex

```diff
@@ -38,15 +48,36 @@ defmodule Shoehorn.Application.EventSupervisor do
     :ok
   end
 
-  defp shutdown(app, reason, s) do
+  defp exited(app, reason, s) do
     {:ok, shutdown_timer_ref} = :timer.apply_after(s.shutdown_timer, :erlang, :halt, [])
-
-    case apply(s.handler, :handle_application, [reason, app, s.handler_state]) do
-      {:halt, _handler_state} ->
+    try do
+      case apply(s.handler, :application_exited, [app, reason, s.handler_state]) do
+        {:halt, _handler_state} ->
+          :erlang.halt()
+        {:continue, handler_state} ->
+          :timer.cancel(shutdown_timer_ref)
+          %{s | handler_state: handler_state}
+      end
+    rescue
```

Comment:
@adkron can correct me, but I think you can add a rescue clause directly to the the defp block

```elixir
defp foo(bar) do
   maybe_go_boom()
rescue
  _ -> IO.puts "that didn't go so well"
end
```